Search repository tags using --list-tags (#7836)
containers/podman: docs/source/markdown/podman-search.1.md

| .Stars          | Star count of image          |
| .Official       | "[OK]" if image is official  |
| .Automated      | "[OK]" if image is automated |

> | .Tag            |Repository tag                |

Reviewer: ```suggestion
| .Tag            | Repository tag               |
```

number of results will be 20, 10 from each (if there are at least 10 matches in each).
The order of the search results is the order in which the API endpoint returns the results.

**--list-tags**

> Search tags available in the repository. Note: the result only contains the Image name and its tag.

Reviewer: As we can use search on `registry.redhat.io/` and get all of the images in the repository:

```suggestion
List the available tags in the repository for each image found. Note: the result contains the Image name and its tag, one line for every tag associated with the image.
```

number of results will be 20, 10 from each (if there are at least 10 matches in each).
The order of the search results is the order in which the API endpoint returns the results.

**--list-tags**

List the available tags in the repository for each image found. Note: --list-tags requires the search term to be a full
repository name. If not specified the registry server, default registries will be searched through (in /etc/containers/registries.conf).  The result contains the Image name and its tag, one line for every tag associated with the image.

> 

Reviewer: This is all very contradictory.  You're saying at the start that you'll list all tags for each image found in the repository, and then that the full name must be used.  I think from our conversations that the image name with the repository must be supplied.  I.e.  you must do `docker.io/library/alpine` and you can't do `docker.io/library/` or `alpine`.  Given that assumption:
```suggestion
List the available tags in the repository for the specified image. **Note:** --list-tags requires the search term to be a fully
specified image name. The result contains the Image name and its tag, one line for every tag associated with the image.
```

Author: fixed

Author: I think the docker.io/library/alpine refer to a repository name, if append a tag to it, it will refer to an image?

Reviewer: @QiWang19 yeah, it's confusing and I may not have helped matters.  When you push a container image to a registry, it creates a repository from it.  How's this?

```
List the available tags in the repository for the specified image. **Note:** --list-tags requires the search term to be a fully
specified image. The result contains the Image name and its tag, one line for every tag found in the repository.
```

Reviewer: :+1: 
nicely done @QiWang19

fedoraproject.org   registry.fedoraproject.org/f25/kubernetes-scheduler                          0
fedoraproject.org   registry.fedoraproject.org/f25/mariadb                                       0
```

```
$ podman search --list-tags  registry.redhat.io/rhel
NAME                      TAG
registry.redhat.io/rhel   7.3-74
registry.redhat.io/rhel   7.6-301
registry.redhat.io/rhel   7.1-9
registry.redhat.io/rhel   7.3-89
registry.redhat.io/rhel   7.3-66
registry.redhat.io/rhel   7.5-424
registry.redhat.io/rhel   7.5-245.1527091554
registry.redhat.io/rhel   7.3-82
registry.redhat.io/rhel   7.4-129
registry.redhat.io/rhel   7.4-120
registry.redhat.io/rhel   7.6-122
registry.redhat.io/rhel   7.7-481
registry.redhat.io/rhel   7.2
registry.redhat.io/rhel   7.5-409
registry.redhat.io/rhel   7.2-75
registry.redhat.io/rhel   7.8-356
registry.redhat.io/rhel   7.7-348
registry.redhat.io/rhel   7.7
registry.redhat.io/rhel   7.4
registry.redhat.io/rhel   7.5
registry.redhat.io/rhel   7.5-409.1533127727
registry.redhat.io/rhel   7.8
registry.redhat.io/rhel   7.5-404
registry.redhat.io/rhel   7.4-105
registry.redhat.io/rhel   7.2-35

> ```

Reviewer: Although this shows what gets returned, for the sake of brevity, I think I'd just show 3 or 4 lines and then a new line with `...` or `(multiple lines removed for brevity)`

Reviewer: And I'm going back/forth on this.  Do others have thoughts?

Reviewer 2: Yes let's list 3 and then ...
